### PR TITLE
Add volume control to Audio

### LINKED ISSUE #6207: [Gradio 4] Adjust waveforms depending on the sound volume range in `gr.Audio`
- [x] I have searched to see if a similar issue already exists.

For loud audio input, the waveform appears like this, and is almost meaningless. So, it would be nice if the display scale of the waveform could be changed according to the sound volume of the input audio.

### PR DESCRIPTION
**Pull request by hannahblair**
## Description

This PR

- Adds ability to adjust volume of waveform
- Adds audio normalisation
- Allows setting trim region colour
- Ensures we send the file detail to `stop_recording`
- Adds a few tests
- Improves typing in some files

Closes: #6205 
Closes #6207 
Closes: #6201
Closes: #6195

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

**Comment by aliabid94** [comment]
I tried running the reverse_audio demo and it threw an error, seems like it was sending null as the value after recording an audio clip.

**Comment by abidlabs** [comment]
I'm seeing the same error @aliabid94 described:

Recording 2023-11-13 at 14 03 05

The microphone waveform seems to also be moving very fast.

I tested `audio_debugger` -- seing the same issue with the input microphone file being null. Note that if I upload a file or click on an example file, there's no issue.

File: js/audio/shared/VolumeControl.svelte [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	export let currentVolume = 1;
+	export let show_volume_slider = false;
+	export let waveform: WaveSurfer;
+
+	onMount(() => {
+		adjustSlider();
+	});
+
+	const adjustSlider = (): void => {
+		let slider = document.getElementById("volume");
```
Comment (aliabid94) [inline js/audio/shared/VolumeControl.svelte:14]: won't this approach only allow one audio element at a time? we should be binding elements and using references to them: https://svelte.dev/repl/76a65cdbd14f40a5818ab327c7b3105b?version=3.29.4

**Comment by aliabid94** [comment]
played around with it and everything worked well! there was a glitch when I opened two sliders at the same time (the coloring messed up) I believe this is for the reason I commented above. Other than that, works great! Code looked good to me from what I could make of it. Great work @hannahblair

**Review by aliabid94 — APPROVED** [review]

**Pull request merged**